### ISSUE TITLE
Notice: Undefined index: tax_label in woocommerce/packages/woocommerce-blocks/src/BlockTypes/MiniCart.php on line 122

### ISSUE THREAD

**Issue by bwp-codes** [woocommerce/woocommerce-gutenberg-products-block#6548]
## Bug Description

When choosing the Woocommerce Tax Option `Display prices during cart and checkout` -> `Including Tax`, the a.m. notice appears in combination with the Mini-Cart Block.
In contrast, the notice is not present when choosing to display prices `Excluding Tax`.

Code excerpt `woocommerce/packages/woocommerce-blocks/src/BlockTypes/MiniCart.php:120-125`:
```
$label_info = $this->get_tax_label();

var_dump($label_info);

$this->tax_label                         = $label_info['tax_label'];
$this->display_cart_prices_including_tax = $label_info['display_cart_prices_including_tax'];
```

Checking the results of the var_dump of the `$label_info` array for
`Including Tax`:
```
array (size=2)
  'label_including_tax' => string '' (length=0)
  'display_cart_prices_including_tax' => boolean true
```
and
`Excluding Tax`:
```
array (size=2)
  'tax_label' => string '(ex. VAT)' (length=9)
  'display_cart_prices_including_tax' => boolean false
```
**Issue:**
The array key `tax_label` is `undefined` when choosing `Including Tax` as a cart / checkout tax display method.

## To reproduce

1. Install a fresh WordPress + Woocommerce and use an FSE Theme.
3. Add the `Mini Cart` as a header widget.
4. Go to `Woocommerce` -> `Settings` -> `Tax` and choose `Display prices during cart and checkout` -> `Including Tax`
5. Go to Frontend & see error

## Expected behavior

(A) Both tax display settings return the same array keys, so the `tax_label` is always defined as an (empty) string
(B) The block includes a check before trying to set `$this->tax_label` e.g.
`woocommerce/packages/woocommerce-blocks/src/BlockTypes/MiniCart.php:122`:
```
$this->tax_label = isset( $label_info['tax_label'] ) ? $label_info['tax_label'] : '';
```

### WordPress (please complete the following information):

-   Core version: 6.0
-   Gutenberg version: (packaged with WordPress)
-   WooCommerce version: 6.5.1
-   WooCommerce Blocks version: (packaged with Woocommerce)
-   Site language: DE (user setting: EN)
-   Any other plugins installed: No

**Comment by sunyatasattva** [comment]
Hello @bwp-codes! Thank you for your report and amazing reproduction instructions.

However, I wasn't able to reproduce this error. I have tried with both our development version and the one bundled in WooCommerce 6.5.1.

I tried adding the MiniCart both in the header and in a single page. I don't see any error, and when I `var_dump` that same variable you mentioned, I get this:

```
// Below you can see that we are loading the version bundled with core
... /wp-content/plugins/woocommerce/packages/woocommerce-blocks/src/BlockTypes/MiniCart.php:125:
array (size=2)
  'tax_label' => string '(incl. VAT)' (length=11)
  'display_cart_prices_including_tax' => boolean true
```

Could this be a localization issue, perhaps? What language is your site set in currently?

Thanks for any further help reproducing this issue.

**Comment by bwp-codes** [comment]
Hello @sunyatasattva!
Thank you for your quick response!

After investigating further, I found out that it is a combination of `Settings` -> `Tax` -> `Prices entered with tax`: `Yes, I will enter prices inclusive of tax`
and `Settings` -> `Tax` -> `Display prices during cart and checkout`: `Including Tax`.

This causes the `tax_label` array key to be missing, see `.../wp-content/plugins/woocommerce/packages/woocommerce-blocks/src/BlockTypes/MiniCart.php:442-455`:
```
if ( $cart->display_prices_including_tax() ) {
	if ( ! wc_prices_include_tax() ) {
		$tax_label                         = WC()->countries->inc_tax_or_vat();
		$display_cart_prices_including_tax = true;
		return array(
			'tax_label'                         => $tax_label,
			'display_cart_prices_including_tax' => $display_cart_prices_including_tax,
		);
	}
	return array(
		'label_including_tax'               => '',
		'display_cart_prices_including_tax' => true,
	);
}
```
We exit the function with the following `return` satement:
```
return array(
  'label_including_tax'               => '',
  'display_cart_prices_including_tax' => true,
);
```

# Proposed fix
I assume it would be correct to set the `tax_label` (incl. VAT) in this case as well via `$tax_label = WC()->countries->inc_tax_or_vat();` and return it via `MiniCart.php:454-458`
```
	return array(
		'label_including_tax'               => '', // not sure what this is used for later.
		'tax_label'                         => $tax_label,
		'display_cart_prices_including_tax' => true,
	);
```

**Comment by bwp-codes** [comment]
Closed / Reopened: Sorry, misclicked! 🙈

**Comment by Aljullu** [comment]
Hey there @bwp-codes, thanks for providing more details. This was fixed in https://github.com/woocommerce/woocommerce-blocks/pull/6537, so the fix will be included in WC Blocks 7.9.0, to be released in June 20.

**Comment by bwp-codes** [comment]
Hello @Aljullu, thanks for the info!
Will this fix also be included in Woocommerce Core Blocks, or do you recommend installing the WC Blocks Plugin separately?

**Comment by Aljullu** [comment]
> Will this fix also be included in Woocommerce Core Blocks, or do you recommend installing the WC Blocks Plugin separately?

Yes, we just released the fix with WC Blocks 7.8.1, so I would recommend installing it. The fix will come in WC core 6.7, but it will take some weeks until it's released, so if the issue is bugging you right now, I would recommend installing WC Blocks. :slightly_smiling_face:
